default-recommendations/conditional-suggestions.rkt

```diff
@@ -0,0 +1,49 @@
+#lang racket/base
+
+
+(require racket/contract/base)
+
+
+(provide
+ (contract-out
+  [conditional-suggestions refactoring-suite?]))
+
+
+(require (for-syntax racket/base)
+         racket/list
+         rebellion/type/record
+         rebellion/private/static-name
+         resyntax/refactoring-rule
+         resyntax/refactoring-suite
+         resyntax/syntax-replacement
+         syntax/parse)
+
+
+;@------------------------------------------(----------------------------------------------------------
+
+
+(define-syntax-class nested-if-else
+  #:attributes (branches branches-size)
+  #:literals (if)
+  (pattern (if cond then (~or* (~var nested nested-if-else) default))
+           #:with branches #`(NEWLINE [cond then]
+                                      #,@(if (attribute default)
+                                             #'(NEWLINE [else default])
+                                             (attribute nested.branches)))
+           #:attr branches-size (if (attribute default)
+                                    2
+                                    (+ 1 (attribute nested.branches-size)))))
```

Comment:
A few related comments:

- If you make `branches` have an _ellipsis depth_ of 1, then you can write `nested.branches ...` instead of `#,@(attribute nested.branches)` in the `#:with branches` template. To do this, write `#:attributes ([branches 1] branches-size)` instead of `#:attributes (branches branches-size)`, and write `#:with (branches ...)` instead of `#:with branches`. If you do this, it would probably be good to also rename it from `branches` to `branch` since attributes with an ellipsis depth greater than zero (the default is zero) are semantically singular, as they're naming each item in a series of items.

- This would look a little cleaner IMO if written as two separate patterns in the syntax class. Syntax classes can have multiple patterns, and they're treated exactly like all of the patterns together were wrapped in an `~or*`. The first one to match is chosen.

- If you add "pattern" to "Lambda-like Keywords" list under DrRacket > Preferences > Editing > Indenting, you'll get much better indentation for syntax class definitions.

All together, that would look like this:

```scheme
(define-syntax-class nested-if-else
  #:attributes ([branch 1] branches-size)
  #:literals (if)

  (pattern (if cond then nested:nested-if-else)
    #:with (branch ...) #'(NEWLINE [cond then] nested.branch ...)
    #:attr branches-size (+ 1 (attribute nested.branches-size)))

  (pattern (if cond then default)
    #:with (branch ...) #'(NEWLINE [cond then] NEWLINE [else default])
    #:attr branches-size 2))
```